pulsar-broker/src/test/java/org/apache/pulsar/broker/service/BrokerServiceAutoTopicCreationTest.java

```diff
@@ -350,6 +350,37 @@ public void testNotAllowSubscriptionTopicCreation() throws Exception{
 
     }
 
+    @Test
+    public void testAutoCreationNamespaceOverridesSubscriptionTopicCreation() throws Exception {
+        pulsar.getConfiguration().setAllowAutoTopicCreation(false);
+        String topicString = "persistent://prop/ns-abc/non-partitioned-topic" + System.currentTimeMillis();
+        String subscriptionName = "non-partitioned-topic-sub";
+        final TopicName topicName = TopicName.get(topicString);
+        pulsar.getAdminClient().namespaces().setAutoTopicCreation(topicName.getNamespace(),
+                AutoTopicCreationOverride.builder()
+                        .allowAutoTopicCreation(true)
+                        .topicType(TopicType.NON_PARTITIONED.toString())
+                        .build());
+
+        admin.topics().createSubscription(topicString, subscriptionName, MessageId.earliest);
+    }
+
+    @Test
+    public void testAutoCreationNamespaceOverridesSubscriptionPartitionedTopicCreation() throws Exception {
+        pulsar.getConfiguration().setAllowAutoTopicCreation(false);
+        String topicString = "persistent://prop/ns-abc/partitioned-topic" + System.currentTimeMillis();
+        String subscriptionName = "partitioned-topic-sub";
+        final TopicName topicName = TopicName.get(topicString);
+        pulsar.getAdminClient().namespaces().setAutoTopicCreation(topicName.getNamespace(),
+                AutoTopicCreationOverride.builder()
+                        .allowAutoTopicCreation(true)
+                        .topicType(TopicType.PARTITIONED.toString())
+                        .defaultNumPartitions(3)
+                        .build());
+
+        admin.topics().createSubscription(topicString + "-partition-0", subscriptionName, MessageId.earliest);
```

Comment:
Do you want to subscribe a to specific partition?

Reply:
Sorry, I don't understand why change the namespace level policy for topic auto-creation to create partitioned topic with 3 partitions, but here check if able to subscribe to one partition, looks like not able to ensure the partitioned topic has been created?

Reply:
Oh, I see.  This test doesn't really make sense. I removed it.

Reply:
Note that currently it's possible to create a topic partition from the subscription creation without the full partitioned topic being created. Is it OK to be able to do that ?

Reply:
@codelipenghui is it OK for you ?